src/Algebra/Graph/Acyclic/AdjacencyMap.hs

```diff
@@ -0,0 +1,449 @@
+module Algebra.Graph.Acyclic.AdjacencyMap (
+  -- * Data structure
+  AdjacencyMap, aam,
+
+  -- * Basic graph construction primitives
+  empty, vertex, disjointOverlay, disjointConnect, vertices,
+
+  -- * Graph properties
+  isEmpty, edgeList, vertexList, edgeSet, vertexSet,
+  vertexCount, edgeCount, adjacencyList, hasEdge, hasVertex,
+
+  -- * Graph transformation
+  removeVertex, removeEdge, transpose, induce,
+
+  -- * Graph composition
+  box,
+
+  -- * Relational operations
+  transitiveClosure,
+
+  -- * Functions on acyclic graphs
+  topSort,
+
+  -- * Acyclic graph construction methods
+  scc, fromGraph, PartialOrder,
+
+  -- * Miscellaneous
+  consistent,
+  ) where
+
+import Algebra.Graph (Graph, foldg)
+import qualified Algebra.Graph.AdjacencyMap as AM
+import qualified Algebra.Graph.AdjacencyMap.Algorithm as AM
+import qualified Algebra.Graph.NonEmpty.AdjacencyMap as NonEmpty
+import qualified Data.Graph.Typed as Typed
+import Data.Set (Set)
+import Data.Coerce (coerce)
+
+{-| The 'AdjacencyMap' data type represents an acyclic graph.
+All the methods provided in this module are safe and cannot be used
+to produce a cyclic graph.
+
+We define a 'Num' instance as a convenient notation for working
+with graphs:
+
+  > vertexList 0           == [0]
+  > edgeList   0           == []
+  > vertexList (1 + 2)     == [1,2]
+  > edgeList   (1 + 2)     == []
+  > vertexList (1 * 2)     == [1,2]
+  > edgeList   (1 * 2)     == [(1,2)]
+  > vertexList (1 + 2 * 3) == [1,2,3]
+  > edgeList   (1 + 2 * 3) == [(2,3)]
+  > vertexList (1 * 2 + 3) == [1,2,3]
+  > edgeList   (1 * 2 + 3) == [(1,2)]
+
+In the `Num` operations, we only keep edges that align with the order
+of integers. Ie. Edges can only be formed from vertex numbered /a/ 
+to a vertex numbered /b/ if and only if /a/ \< /b/:
+
+  > edgeList (1 * 2) == [(1,2)]
+  > edgeList (2 * 1) == []
+  > (2 * 1)          == (2 + 1)
+
+__Note:__ the 'Num' instance does not satisfy several "customary laws"
+of 'Num', which dictate that 'fromInteger' @0@ and 'fromInteger' @1@
+should act as additive and multiplicative identities, and 'negate' as
+additive inverse. Nevertheless, overloading 'fromInteger', '+' and '*'
+is very convenient when working with algebraic graphs; we hope that
+in future Haskell's Prelude will provide a more fine-grained class
+hierarchy for algebraic structures, which we would be able to utilise
+without violating any laws.
+-}
+newtype AdjacencyMap a = AAM
+  { aam :: AM.AdjacencyMap a
+  } deriving (Show, Eq, Ord)
+
+-- | Check if the internal graph representation is consistent,
+-- i.e. that all edges refer to existing vertices and the graph
+-- is acyclic. It should be impossible to create an inconsistent 
+-- adjacency map.
+--
+-- @
+-- consistent 'empty'                 == True
+-- consistent (vertex x)            == True
+-- consistent (disjointOverlay x y)        == True
```

Comment:
The alignment is off. Also, only `empty` has quotes -- I think all primitives should be quoted. The general rule is: quote all functions apart from the function that is being defined.

Reply:
Oh! That's because I've replaced `overlayD` with `disjointOverlay`. It should be off in `disjointConnect` too. My bad, I'll fix it!